internal/backend/runtime/omni/controllers/omni/internal/kubernetes/versions.go

```diff
@@ -46,34 +54,46 @@ func CalculateUpgradeVersions(ctx context.Context, r controller.Reader, currentV
 		}
 	}
 
-	// skip versions which do not have an upgrade path from the current version
+	// skip versions which are not supported with the current version of Talos
 	for version := range availableVersions {
-		path, err := upgrade.NewPath(currentVersion, version)
+		k8sVersion, err := compatibility.ParseKubernetesVersion(version)
 		if err != nil {
 			return nil, err
 		}
 
-		if !path.IsSupported() {
+		if k8sVersion.SupportedWith(talosVersion) != nil {
 			delete(availableVersions, version)
+			unsupportedVersions = append(unsupportedVersions, version)
 		}
 	}
 
-	// skip versions which are not supported with the current version of Talos
+	// skip versions which do not have an upgrade path from the current version
 	for version := range availableVersions {
-		k8sVersion, err := compatibility.ParseKubernetesVersion(version)
+		path, err := upgrade.NewPath(currentVersion, version)
 		if err != nil {
 			return nil, err
 		}
 
-		if k8sVersion.SupportedWith(talosVersion) != nil {
+		if !path.IsSupported() {
```

Comment:
I feel this might be a really long list, while all we need to show to the user is that there is a next version after supported ones, which is not available for one of two reasons. I don't think anybody needs a list of a hundred unavailable versions?

Reply:
I agree, we shouldn't show everything with notes about why. We should only show the available options for their version of Talos. It would be even better if there were an "Upgrade to latest" button that would automatically upgrade to whatever latest version they're allowed to upgrade to would be.

Reply:
My point is to avoid bloating resources with a list of hundreds unsupported versions (in theory), while we only need to show the _next_ one grayed out. We don't need other versions.

E.g. if I'm on 1.33.7, and my Talos only supports up to 1.33.x, then 1.34.0 is grayed out "unsupported by Talos".

If I'm on 1.32, Talos supports up to 1.34.x, I can upgrade to 1.33, but 1.34.0 is grayed out as "upgrade to 1.33.x first".

We could also show previous one grayed out saying that we don't support downgrades.

Reply:
I'm ok with that also. I saw there are min/max versions in the code that it uses to generate these versions, are those returned somewhere already? I could use them to generate the version you are saying. If not returned yet I can try return them.